### PR TITLE
Don't mark machine ready=false when internalIP is known

### PR DESCRIPTION
**Pull request by davidvossel**
We've seen instances where the vmi.Status.Interfaces list temporarily clears and stops reporting the default instance IP. This occurs when the qemu guest agent can not be contacted, such as during an internal soft reboot. It also occurs for reasons we do not 100% understand yet.

Since we know this vmi.Status.Interfaces field is dependent on the qemu guest agent, there will be times where the agent is unavailable. To smooth over those time periods, capk now only reports IP changes on the node when we're certain the IP has actually changed, and does not clear out the internal IP when no default ip is reported on the vmi status.

```release-note
NONE
```

### CODE REVIEW

File: controllers/kubevirtmachine_controller.go [hunk 1]
```diff
@@ -349,6 +359,15 @@ func (r *KubevirtMachineReconciler) reconcileNormal(ctx *context.MachineContext)
 	return ctrl.Result{}, nil
 }
 
+func machineHasKnownInternalIP(kubevirtMachine *infrav1.KubevirtMachine) bool {
+	for _, addr := range kubevirtMachine.Status.Addresses {
+		if addr.Type == clusterv1.MachineInternalIP && addr.Address != "" {
```
Comment (qinqon) [inline controllers/kubevirtmachine_controller.go:364]: we should skip ipv6 link local

**Review by qinqon — CHANGES_REQUESTED** [review]

File: controllers/kubevirtmachine_controller.go (hunk 1 above)
Comment (davidvossel) [reply controllers/kubevirtmachine_controller.go:364]: That made sense to me at first, but after discussing this further I'm not 100% sure. It's theoretically possible someone might be using this in a network where link local addresses for VMs technically work.  It's just that the use case we're most familiar with (vms on pod network) wouldn't use link local.

**Pull request merged**